### PR TITLE
Added capability to query last successfully built version

### PR DESCRIPTION
**Pull request by johannesduesing**
Added the ability to query the backend for the last successfully built version for a particular collection, as stated in issue #83. 
Since i need built versions to test my feature, i introduced a UnitTest that builds a version of a specific project via REST interface. This test does pass, but takes a lot of time (about 1~2 mins) and won't scale well, especially since i would like to extend the testing to catch more corner cases. Is this generally speaking the right way to go, or is there a better way to mock successful build results without actually building a project?

### CODE REVIEW

File: bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
 
+
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java]: nit: extra line not needed

File: bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java [hunk 2]
```diff
@@ -150,6 +152,42 @@ public void deleteCollection(String id) throws IOException {
         // delete the collection
         collectionDao.delete(id);
     }
+     
+    public VersionDTO getLastSuccessfullyBuiltVersion(String collectionID) throws IOException{
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java]: nit: space before the '{'

File: bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java [hunk 3]
```diff
@@ -150,6 +152,42 @@ public void deleteCollection(String id) throws IOException {
...
+    	authorizer.requireRole("RegisteredUser");
+    	
+        // make sure the user is the owner of the collection
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java]: nit: Comments start with a capital letter and end with a '.'

File: bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java [hunk 4]
```diff
@@ -150,6 +152,42 @@ public void deleteCollection(String id) throws IOException {
...
+        	BuildResultDTO buildResult = buildResultDao.findByVersion(version.id);
+        	
+            if(buildResult != null && buildResult.status.equals("FINISHED")){
+            	successfullyBuiltVersions.add(version);
+            }        	
+        }
+        
+        successfullyBuiltVersions.sort(new java.util.Comparator<VersionDTO>() {
+
+			@Override
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.app/src/de/fraunhofer/abm/app/controllers/CollectionController.java]: nit: fix indentations.

File: bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java [hunk 5]
```diff
@@ -173,4 +173,94 @@ public void testCollectionById() throws IOException {
         response = HttpUtils.delete(uri, headers, charset);
         Assert.assertEquals(200, response.getResponseCode());
     }
+    
+    @Test
+    public void testLastSuccessfullyBuiltVersion() throws IOException{
+    	//Create a private collection first
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java]: nit: space after "//".

File: bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java [hunk 6]
```diff
@@ -173,4 +173,94 @@ public void testCollectionById() throws IOException {
...
+        response = HttpUtils.post(uri, headers, payload.getBytes(charset), charset);
+        Assert.assertEquals(200, response.getResponseCode());
+        
+        //Get BuildDTO
+        uri = baseUri + "/rest/builds/"+USER;
+        
+        float build_progress = 0F;
+        while(build_progress < 1F){
+        	String json = HttpUtils.get(uri, headers, charset);
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java]: nit: indentation.

File: bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java [hunk 7]
```diff
@@ -173,4 +173,94 @@ public void testCollectionById() throws IOException {
...
...-maven-app/releases{/id}\",\"description\":\"For an introductory tutorial on how to use Jenkins to build a simple Java application with Maven.\",\"contributorsUrl\":\"https://api.github.com/repos/jenkins-docs/simple-java-maven-app/contributors\",\"isPrivate\":false,\"commitsUrl\":\"https://api.github.com/repos/jenkins-docs/simple-java-maven-app/commits{/sha}\",\"openIssues\":9,\"latestUpdate\":\"2018-05-21T09:01:28Z\",\"score\":188,\"starred\":24,\"id\":\"a38166a0-0661-3f51-8085-f22c1b163f78\",\"forks\":2216,\"owner\":\"jenkins-docs\",\"hasWiki\":true,\"defaultBranch\":\"master\",\"htmlUrl\":\"https://github.com/jenkins-docs/simple-java-maven-app\",\"creationDate\":\"2017-09-26T02:35:06\",\"contentsUrl\":\"https://api.github.com/repos/jenkins-docs/simple-java-maven-app/contents/{+path}\",\"remoteId\":104826554,\"repositoryUrl\":\"https://github.com/jenkins-docs/simple-java-maven-app.git\",\"hasDownloads\":true,\"license\":\"\",\"watched\":24,\"size\":13,\"name\":\"simple-java-maven-app\",\"repositoryType\":\"git\",\"properties\":[]}}],\"filtered\":false,\"frozen\":false,\"comment\":\"Initial Version\",\"creationDate\":\"2018-05-28T13:15:19\",\"collectionId\":\""+collectionid+"\"}";
```
Comment (nguyenLisa) [inline bnd-workspace/de.fraunhofer.abm.test.http/src/de/fraunhofer/abm/test/http/CollectionControllerTest.java]: nit: wrap line at 120 characters.
